Entities.js

```diff
@@ -0,0 +1,339 @@
+  }
+  if(Math.abs(this.momentum[1] + vertical/4) > 4){
+    this.momentum[1] = Math.sign(this.momentum[1]) * 4
+  }
+  else{
+    this.momentum[1] += vertical/4
+  }
+  //slow the knights velocity by a small amount (to lessen drifting)
+  this.momentum[0] -= .01 * Math.sign(this.momentum[0])
+  this.momentum[1] -= .01 * Math.sign(this.momentum[1])
+
+  this.x += this.momentum[0]
+  this.y += this.momentum[1]
+  //update the knights position and record the change
+  GlobalX = this.x
+  GlobalY = this.y
+}
+}
+
+
+class Brick {
+  constructor(id) {
+    this.id = id;
+    let x = Math.random()*500;
+    let y = 5;
+    this.x = x;
+    this.y = y;
+    this.backgroundColor = 'red';
+    this.height = 75;
+    this.width = 75;
+    this.Image = "./assets/Brick.jpeg";
+    this.renderer = <CreateBrick/>;
+    let randomNum = Math.random()*3;
+    this.velocityX = randomNum;
+    this.velocityY = 3 - randomNum;
+  }
+
+  update() {
+    //Bouncing off walls:
+    if(this.x >= (MAXwidth - this.width)){
+      this.reverse(-1,1)
+    }
+    if(this.x <= 0){
+      this.reverse(-1,1)
+    }
+    if(this.y >= (MAXheight - this.height)){
+      this.reverse(1,-1)
+    }
+    if(this.y <= 0){
+      this.reverse(1,-1)
+    }
+
+  //check for player intersections  
+    if(this.y + this.width < GlobalY  || this.y > GlobalY + playerHeight){
+    }
+    else if(this.x > GlobalX + playerWidth || this.x + this.width < GlobalX ){
+
+    }
+    else{
+      DeleteEntity(this.id)
+      counter += 1
+    }
+
+//updating the bricks position
+    this.x += this.velocityX;
+    this.y += this.velocityY;
+  }
+// TODO learn to use matterjs to handle collisions and possibly movement
+  reverse(xMult, yMult) {
+    this.velocityX = (this.velocityX * xMult)
+    this.velocityY = (this.velocityY * yMult)
+  }
+}
+
+class Circle {
+  constructor(id){
+    this.id = id;
+    let x = 5
+    let y = Math.random()*500
+    this.borderRadius = 75;
+    this.x = x;
+    this.y = y;
+    this.height = 75;
+    this.width = 75;
+    this.renderer = <CreateCircle/>
+    let randomNum = Math.random()*3
+
+    this.velocityX = randomNum;
+    this.velocityY = 3 - randomNum;
+    this.backgroundColor = 'red';
+    }
+  
+  update() {
+    // change velocity to move towards knight
+    this.velocityX += (GlobalX - this.x)/5000
+    this.velocityY += (GlobalY - this.y)/5000
+    //move
+    this.x += this.velocityX
+    this.y += this.velocityY
+
+    //check for intersection with walls
+    if(this.x >= (MAXwidth - this.width)){
+      this.reverse()
+    }
+    if(this.x <= 0){
+      this.reverse()
+    }
+    if(this.y >= (MAXheight - this.height)){
+      this.reverse()
+    }
+    if(this.y <= 0){
+      this.reverse()
+    }
+    
+//check for intersection with knight
+//check if there is a possible for up down
+//check if there is a possible for right left
+//check if those two things overlap
+    if(this.y + this.width < GlobalY  || this.y > GlobalY + playerHeight){
```

Comment:
"Intersection" is a good candidate for a common function somewhere.
Although as we talked about, you might need to treat circles/rectangles/other shapes differently.